src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -654,8 +654,9 @@ private void Get_CORINFO_SIG_INFO(MethodSignature signature, CORINFO_SIG_INFO* s
             sig->sigInst.methInst = null; // Not used by the JIT
             sig->sigInst.methInstCount = (uint)signature.GenericParameterCount;
 
-            sig->pSig = (byte*)ObjectToHandle(signature);
+            sig->pSig = null;
             sig->cbSig = 0; // Not used by the JIT
+            sig->methodSignature =(MethodSignatureInfo*)ObjectToHandle(signature);
```

Comment:
I think at this point it's worth adding overrides for `ObjectToHandle` and `HandleToObject` that implement the casts to and from `MethodSignatureInfo*` and `MethodSignature` respectively.

Crossgen2 already does that for a number of the opaque handle types, so I don't see a reason to not add another override for this for simplicity. (Also adding one for `CORINFO_SCOPE_HND <-> MethodIL` would be good.)

Reply:
Good idea - I cleaned up the casts around `ObjectToHandle` and vice versa. There are a couple remaining casts to deal with the fact that sometimes we pass a `LocalVariableDefinition[]` instead of a `MethodSignature` in the handle field.

Reply:
It looks good to me!